theSrc/scripts/LabeledScatter.js

```diff
@@ -414,6 +443,90 @@ class LabeledScatter {
+    const text_anchor = config.titleAlignment === 'Left' ? 'start' : (config.subtitleAlignment === 'Center' ? 'middle' : 'end')
+    title_element
+      .attr('x', x)
+      .attr('dy', 0)
+      .style('alignment-baseline', 'text-before-edge')
+      .style('text-anchor', text_anchor)
+
+    title_element
+      .selectAll('tspan')
+      .attr('x', x)
+      .style('alignment-baseline', 'text-before-edge')
+      .style('text-anchor', text_anchor)
+
+    const subtitle_element = this.getAnnotationElement('subtitle', plotly_chart_layout)
+    if (subtitle_element !== null) {
+      const subtitle_x = config.subtitleAlignment === 'Left' ? 0 : (config.subtitleAlignment === 'Center' ? 0.5 * this.width : this.width)
+      const subtitle_text_anchor = config.subtitleAlignment === 'Left' ? 'start' : (config.subtitleAlignment === 'Center' ? 'middle' : 'end')
+      subtitle_element
+        .select('.cursor-pointer')
+        .attr('x', 0)
+        .attr('y', 0)
+        .attr('transform', `translate(${subtitle_x},${this.titleBottom(config)})`)
+      subtitle_element
+        .select('.annotation-text')
+        .attr('x', 0)
+        .attr('y', 0)
+        .style('alignment-baseline', 'text-before-edge')
+        .style('text-anchor', subtitle_text_anchor)
+      subtitle_element.selectAll('.annotation-text tspan').attr('x', 0)
+    }
+
+    const xtitle_element = this.getAnnotationElement('xtitle', plotly_chart_layout)
+    if (xtitle_element !== null) {
+      xtitle_element
+        .select('.annotation-text-g')
+        .attr('transform', `translate(${0.5 * this.width},${this.height})`)
+
+      xtitle_element
+        .select('.cursor-pointer')
+        .attr('transform', null)
+
+      xtitle_element
+        .select('.annotation-text')
+        .attr('x', 0)
+        .attr('y', 0)
+        .attr('alignment-baseline', 'text-after-edge')
+    }
+
+    const ytitle_element = this.getAnnotationElement('ytitle', plotly_chart_layout)
+    if (ytitle_element !== null) {
+      ytitle_element
+        .select('.annotation-text-g')
+        .attr('transform', `translate(0,${0.5 * this.height}) rotate(-90)`)
+
+        ytitle_element
+          .select('.cursor-pointer')
+          .attr('transform', null)
+
+        ytitle_element
+          .select('.annotation-text')
+          .attr('x', 0)
+    }
+  }
+
+  getAnnotationElement (name, plotly_chart_layout) {
+    const index = plotly_chart_layout.annotations.map(a => a.name).indexOf(name)
+    if (index === -1) {
+      return null
+    }
+    const annotations = d3.select(this.rootElement).selectAll('.annotation')
+    return annotations[0][0] ? d3.select(annotations[0][index]) : null
+  }
+
+  titleBottom (config) {
+    if (config.title && config.title.length > 0) {
+      return config.title.split('<br>').length * config.titleFontSize * LINE_HEIGHT_AS_PROPORTION_OF_FONT_SIZE
```

Comment:
This seems to be the only place you use `LINE_HEIGHT_AS_PROPORTION_OF_FONT_SIZE` in LabeledScatter.js? Maybe these functions should go in to PlotlyChartElements.js and you just export them across?

Reply:
Yes, I've moved `titleBottom` to PlotlyChartElements.js in 112c17f24bfb8460afe6351f6fc3f3c2a58a9843

Reply:
Maybe the function name should be `titleHeight` or `titleVerticalSpace` instead?